src/fu-tool.c

```diff
@@ -749,13 +750,30 @@ fu_util_install (FuUtilPrivate *priv, gchar **values, GError **error)
 	if (!fu_engine_install_tasks (priv->engine, install_tasks, blob_cab, priv->flags, error))
 		return FALSE;
 
+	/* we don't want to ask anything */
+	if (priv->no_reboot_check) {
+		g_debug ("skipping reboot check");
+		return TRUE;
+	}
+
+	/* show reboot if needed */
+	for (guint i = 0; i < devices_possible->len; i++) {
+		FwupdDevice *dev = g_ptr_array_index (devices_possible, i);
+		if (fwupd_device_has_flag (dev, FWUPD_DEVICE_FLAG_NEEDS_SHUTDOWN))
+			requires_shutdown = TRUE;
+		else if (fwupd_device_has_flag (dev, FWUPD_DEVICE_FLAG_NEEDS_REBOOT))
+			requires_reboot = TRUE;
+	}
+
 	/* success */
-	return TRUE;
+	return fu_util_prompt_complete (requires_shutdown, requires_reboot, error);
```

Comment:
I think you should pass a bit-field of `FwupdDeviceFlags` to `fu_util_prompt_complete` -- you can build that using `flags |= FWUPD_DEVICE_FLAG_NEEDS_SHUTDOWN` etc.